mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/LinkBankAccountViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package org.mifos.mobilewallet.mifospay.bank.presenter
+
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import androidx.lifecycle.ViewModel
+import com.mifos.mobilewallet.model.domain.BankAccountDetails
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import org.json.JSONObject
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.common.Constants
+import org.mifos.mobilewallet.mifospay.domain.model.Bank
+import org.mifos.mobilewallet.mifospay.utils.FileUtils
+import java.util.Random
+import javax.inject.Inject
+
+
+@HiltViewModel
+class LinkBankAccountViewModel @Inject constructor() : ViewModel() {
+
+
+    private val _popularBanks = MutableStateFlow<List<Bank>>(emptyList())
+    var popularBanks = _popularBanks.asStateFlow()
+
+    private val _allBanks = MutableStateFlow<List<Bank>>(emptyList())
+    var allBanks = _allBanks.asStateFlow()
+
+    private val _filteredBanks = MutableStateFlow<List<Bank>>(emptyList())
+    var filteredBanks = _filteredBanks.asStateFlow()
+
+    private val _bankState = MutableStateFlow<BankUiState>(BankUiState.Loading)
+    val bankState = _bankState.asStateFlow()
+
+
+    fun getAllBanks(context: Context) {
```

Comment:
fetch flowable banks list from here 
<img width="1728" alt="image" src="https://github.com/openMF/mobile-wallet/assets/5724482/fea1cc8a-dadb-40da-8ace-d16f985f2110">

You can take an example 
<img width="1728" alt="image" src="https://github.com/openMF/mobile-wallet/assets/5724482/be479406-b7f3-4cc5-a5a0-423a3f3a0412">

You can have a query field 

```
    private val _bankSearchquery = MutableStateFlow<String>>("")
    var bankSearchquery = _bankSearchquery.asStateFlow()
    
fun updateBankSearchquery(query: String) {
_bankSearchquery.update { query }
} 

 val banks: StateFlow<List<Bank>> = combine(
        localAssetRepository.getBanks(),
        bankSearchquery,
        ::Pair
    )
        .map { 
            val countries = it.first
            signupData = signupData.copy(
                countryId = countries.find { it.name == signupData.countryName }?.id ?: ""
            )
           val allBanks =  it.first.map { Bank (it, R.drawable.ic_bank, 1)}.filter { it.name.contains(it.second) }
            
             popularBanks = // add isPopular field in Bank set them above true if below bank name matches so you can filter with single source and show on ui
             
            LoadedBankState(
              allBanks = allBanks, 
       
           )
        }
        .stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(5_000),
            initialValue = emptyList(),
        )
```

Reply:
using this implementation you don't need other fields as user type and you update query, it will automatically update the bank list.

Reply:
@therajanmaurya here on init i am getting the banks like this. The popular banks is a separate thing in ui and some of the popular banks is not present in json. the popular banks have their own drawable as u can see
![Screenshot (319)](https://github.com/openMF/mobile-wallet/assets/91717339/f14d83f6-243b-4c97-8abe-44a509baaa78)

After that as per your search logic i am setting the filtered banks and the popular banks from previous state
![Screenshot (320)](https://github.com/openMF/mobile-wallet/assets/91717339/de6ec592-c4ad-4851-be25-d32186ec40ef)

Please let me know if any changes required.